### PR TITLE
Add docs for QUIC configuration

### PR DESCRIPTION
**Pull request by maskit**

### CODE REVIEW

File: doc/admin-guide/files/records.config.en.rst [hunk 1]
```diff
@@ -3721,6 +3734,240 @@ HTTP/2 Configuration
    |TS| gracefully closes connections that have stream error rates above this
    setting by sending GOAWAY frames.
 
+HTTP/3 Configuration
+====================
+
+There is no configuration on this release.
```
Comment (masaori335) [inline doc/admin-guide/files/records.config.en.rst]: Do you ignore `proxy.config.http3.*` intentionally?

Comment (maskit) [reply doc/admin-guide/files/records.config.en.rst]: Yes. I don’t remember the exact reason, but they didn’t seem like something we want to show to users at the moment. I can add them if you want.

Comment (masaori335) [reply doc/admin-guide/files/records.config.en.rst]: These configs are knobs for HTTP/3 SETTINGS Parameters. It looks very similar to what we have with HTTP/2.

HTTP/3 is unstable yet, so I'm fine with no docs for them right now. But it'd be better if the sentence is something like TBD or nothing.

File: doc/admin-guide/files/records.config.en.rst [hunk 2]
```diff
@@ -3721,6 +3734,240 @@ HTTP/2 Configuration
...
+
+   A size of hash table that stores connection information.
+
+.. ts:cv:: CONFIG proxy.config.quic.proxy.config.quic.num_alt_connection_ids INT 65521
+   :reloadable:
+
+   A number of alternate Connection IDs that |TS| provides to a peer. It has to
+   be at least 8.
+
+.. ts:cv:: CONFIG proxy.config.quic.stateless_retry_enabled INT 0
```
Comment (masaori335) [inline doc/admin-guide/files/records.config.en.rst]: Kind of off-topic. Should we rename this `proxy.config.quic.server.stateless_retry_enabled` ?

File: doc/admin-guide/files/records.config.en.rst (hunk 1 above)
Comment (maskit) [reply doc/admin-guide/files/records.config.en.rst]: I just remember the reason. I don't think the settings work now.

File: doc/admin-guide/files/records.config.en.rst (hunk 2 above)
Comment (maskit) [reply doc/admin-guide/files/records.config.en.rst]: Maybe. `proxy.config.quic.stateless_retry_enabled_in` can be another option.

File: doc/admin-guide/files/records.config.en.rst (hunk 1 above)
Comment (maskit) [reply doc/admin-guide/files/records.config.en.rst]: I slightly changed the sentence. Feel free to change it if it's still not enough.

**Pull request merged**